beetsplug/spotify.py

```diff
@@ -683,11 +682,9 @@ def _fetch_info(self, items, write, force):
             if write:
                 item.try_write()
 
-    def track_info(self, track_id=None):
+    def track_info(self, track_id: str):
         """Fetch a track's popularity and external IDs using its Spotify ID."""
-        track_data = self._handle_response(
-            requests.get, self.track_url + track_id
-        )
+        track_data = self._handle_response("get", self.track_url + track_id)
```

Comment:
Seems like there should be an early return here if 'track_data is None`?

Reply:
Oh, what happened here is that I thought that the mypy warnings below are due to the possibility of `track_data` being `None`: As you say, that's not actually true. What mypy really seems to be complaining about is nested accesses. Not sure whether those should have more validation, but that's probably out-of-scope in this PR. (Is there some higher-level function that would catch the exception if `track_data` wouldn't match the expected schema?)

Reply:
I don't think we have any validation in place. We just hope for the best and pray I guess 😅